tests/test_state.py

```diff
@@ -3,37 +3,40 @@
 
 import pytest
 from benchcab.utils.state import State, StateAttributeError
+from tempfile import TemporaryDirectory
 
 
-@pytest.fixture()
-def state():
-    """Return a State object."""
-    return State(state_dir=Path("my_state"))
```

Comment:
Have you considered using [`yield` fixtures](https://docs.pytest.org/en/6.2.x/fixture.html#yield-fixtures-recommended) instead of adding `with TemporaryDirectory() as tmp_dir` to the tests?

Reply:
Yes, but the `with` approach appears to be more reliable when acting on temp directories and files.